pkg/ottl/e2e/e2e_test.go

```diff
@@ -1084,6 +1084,30 @@ func Test_e2e_converters(t *testing.T) {
 				m.PutInt("bar", 5)
 			},
 		},
+		{
+			statement: `set(attributes["test"], {"list":[{"foo":"bar"}]})`,
```

Comment:
Can you add some more e2e tests using map literals and slice literals in other functions? That would give me confidence that we're supporting `map[string]any`/`pcommon.Map` and `[]any`/`pcommon.Slice` consistently across the framework.

Reply:
sure, will add a couple of tests today

Reply:
alright, I have added some more test cases for several functions making use of map structures.
I have also added another test type to the e2e tests that can be used to test a statement sequence to also be able to test the editor functions with map literals